install_config/persistent_storage/persistent_storage_csi.adoc

```diff
@@ -0,0 +1,408 @@
+
+[source,shell]
+----
+# oc new-project csi
+Now using project "csi" on server "https://example.com:8443".
+
+# oc create serviceaccount cinder-csi
+serviceaccount "cinder-csi" created
+
+# oc adm policy add-scc-to-user privileged system:serviceaccount:csi:cinder-csi
+scc "privileged" added to: ["system:serviceaccount:csi:cinder-csi"]
+----
+
+2. Apply this yaml file to create DaemonSet with external attacher, Deployment with external provisioner and DaemonSet with the CSI driver.
+
+[source,yaml]
+----
+# This YAML file contains all API objects that are necessary to run Cinder CSI
+# driver.
+#
+# In production, this needs to be in separate files, e.g. service account and
+# role and role binding needs to be created once, while StatefulSet set may
+# require some tuning.
+#
+# It server as an example how to use external attacher and external provisioner
+# images shipped with {product-title} with a community CSI driver.
+
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cinder-csi-role
+rules:
+  - apiGroups: [""]
+    resources: ["persistentvolumes"]
+    verbs: ["create", "delete", "get", "list", "watch", "update"]
+  - apiGroups: [""]
+    resources: ["persistentvolumeclaims"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: [""]
+    resources: ["nodes"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["storageclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["volumeattachments"]
+    verbs: ["get", "list", "watch", "update"]
+
+---
+
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cinder-csi-role
+subjects:
+  - kind: ServiceAccount
+    name: cinder-csi
+    namespace: csi
+roleRef:
+  kind: ClusterRole
+  name: cinder-csi-role
+  apiGroup: rbac.authorization.k8s.io
+
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: cinder-csi
+  labels:
+    app: cinder-csi
+spec:
+  selector:
+    app: cinder-csi
+  ports:
+    - name: dummy
+      port: 12345
+
+---
+apiVersion: v1
+data:
+  cloud.conf: W0dsb2JhbF0KYXV0aC11cmwgPSBodHRwczovL2NpLXJob3MuY2VudHJhbGNpLmVuZy5yZHUyLnJlZGhhdC5jb206MTMwMDAvdjIuMC8KdXNlcm5hbWUgPSBvcGVuc2hpZnQtcWUtamVua2lucwpwYXNzd29yZCA9IHV2W3NWODBYbgp0ZW5hbnQtaWQgPSBlMGZhODViNmEwNjQ0Mzk1OWQyZDNiNDk3MTc0YmVkNgpyZWdpb24gPSByZWdpb25PbmUK
+kind: Secret
+metadata:
+  creationTimestamp: null
+  name: cloudconfig
+---
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: cinder-csi-provisioner
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: cinder-csi-provisioner
+  template:
+    metadata:
+      labels:
+        app: cinder-csi-provisioner
+    spec:
+      serviceAccount: cinder-csi
+      containers:
+        - name: csi-external-provisioner
+          image: registry.reg-aws.openshift.com:443/openshift3/csi-provisioner:v3.10
```

Comment:
Maybe we should use the product registry: registry.access.redhat.com?

Reply:
fixed